crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -213,38 +212,41 @@ where
         let address_filter = FilteredParams::address_filter(&filter.address);
         let topics_filter = FilteredParams::topics_filter(&topics);
 
+        // loop over the range of new blocks and check logs if the filter matches the log's bloom
+        // filter
         for block_number in from_block..=to_block {
             if let Some(block) = self.client.block_by_number(block_number).to_rpc_result()? {
                 // only if filter matches
                 if FilteredParams::matches_address(block.header.logs_bloom, &address_filter) &&
                     FilteredParams::matches_topics(block.header.logs_bloom, &topics_filter)
                 {
-                    self.append_matching_block_logs(&mut logs, &filter_params, block);
-
-                    // TODO size check
+                    // get receipts for the block
+                    if let Some(receipts) =
+                        self.client.receipts_by_block(block.number.into()).to_rpc_result()?
+                    {
+                        let block_hash = block.hash_slow();
```

Comment:
why is hash re-calculated here? it should be accessible on the `block`

Reply:
it does not include the hash, it just includes the header.

Reply:
seems a bit redundant since we do select the hash when we query for the block
https://github.com/paradigmxyz/reth/blob/a0ff24b691aee859db10a11b48df56100fd5e1e2/crates/rpc/rpc/src/eth/api/block.rs#L63-L66

Reply:
this is an entirely different function, that is not called here.

Reply:
but looks like there should be a provider function that also returns the block hash when it fetches the block.

Reply:
Not a blocker for this PR, we should profile all of these anyway for various queries